lib/ansible/executor/module_common.py

```diff
@@ -442,18 +442,46 @@ def _get_shebang(interpreter, task_vars, templar, args=tuple()):
        file rather than trust that we reformatted what they already have
        correctly.
     """
-    interpreter_config = u'ansible_%s_interpreter' % os.path.basename(interpreter).strip()
+    interpreter_name = os.path.basename(interpreter).strip()
 
-    if interpreter_config not in task_vars:
-        return (None, interpreter)
+    # FUTURE: add logical equivalence for python3 in the case of py3-only modules
 
-    interpreter = templar.template(task_vars[interpreter_config].strip())
-    shebang = u'#!' + interpreter
+    # check for first-class interpreter config
+    interpreter_config_key = "DEFAULT_INTERPRETER_%s" % interpreter_name.upper()
+
+    if C.config.get_configuration_definitions().get(interpreter_config_key):
```

Comment:
It looks like this conditional is slightly off?  If ansible_python_interpreter is set, then it needs to use that.  But in this code, it looks like the value of DEFAULT_INTERPRETER_PYTHON causes that to be ignored.

Reply:
I believe based on what I understand, that the later `get_config_value` in which we pass `task_vars` will sort this out.  I have not test that however.

Reply:
This is to determine if the interpreter has config definitions present for discovery at all, not what their values are. For Python, this should always be true (which delegates the handling of `ansible_python_interpreter` value to the config system- see the config definition for `DEFAULT_INTERPRETER_PYTHON`), for anything else, it drops into the `else` that uses the old behavior of just consulting the task vars.